### ISSUE TITLE
AI development is so fast

### ISSUE THREAD

**Issue by tak2hu** [Snowad14/Novel2Anime#1]
Dang why have I never thought of this, it’s interesting.

With [LangChain](https://python.langchain.com/en/latest/modules/indexes/getting_started.html), it makes it easier to extract dialogues, probably. It’s a good starting point.

Haven’t really tried anything, just want to mention that here.

I wonder if it’s possible to use nanogpt or models that can run locally and use langchain.

**Comment by Snowad14** [comment]
Yes the development is quite fast, I am more impatient with the development of Text2Video which is very promising.
There are also projects to create voices like ElevenLabs locally, but researchers are reluctant to take risks.

**Comment by tak2hu** [comment]
Well working on the first step, to extract text, is a good start. So that instead of sending a whole chapter or even volume(s) to a model is computationally less expensive.

I think LLMs are overkill to extract, summarize, dan classify text in novels. But also it’s the most straightforward approach 

Maybe a simple algorithm could do for dialogues (i.e regex for quotation marks) but I’m curious if a very light model exists that can be the base for text extraction.

Summarize: to get the story line/structure/timeline/flow
Classify: who are the characters, who’s talking, what action is a character doing, where is the character (label places), maybe even set which role is who (protag, sub char, antagonist, etc) and relationships of each character or which organization does a character belong…
Sentiment Analysis: is the dialogue sad? (To style the prompt for text-to-video or text-image)

**Comment by Snowad14** [comment]
The hardest part is probably to make a prompt according to the text, but with the new text2Video models it might be a bit easier.

Also, an algorithm will never be able to detect who is speaking, it's a much harder task than you think and even GPT-4 can be wrong. In addition, a GPT-4 (or good futures open source model) will be able to analyze all of them at the same time without the need for another model

**Comment by tak2hu** [comment]
Btw [have you seen this?](https://www.reddit.com/r/AnimeResearch/comments/12qdxl3/animov01_highresolution_anime_finetune_of/), some scenes look uncanny lol.

Besides making videos from novels, it’s desirable to produce manga, visual novel (renpy), or even a game (godot).

Also [this looks cool](https://github.com/ShuhongChen/panic3d-anime-reconstruction)

**Comment by Snowad14** [comment]
Yes I know, I follow a lot the development of Text2Video but we are still at least one year away from having a real good quality
